### ISSUE TITLE
Force absolute path for some items

### ISSUE THREAD

**Issue by TheElectronWill** [rust-lang/rust-clippy#14520]
### What it does

Lint `absolute_paths` (#10568) warns about items that use an absolute path instead of a `use`.
Clippy misses the opposite lint: **warn about unqualified paths** that are imported where they should not.

### Advantage

An example is the `log` crate, which is often used as `log::info!` or `log::error!`. Unfortunately, newcomers may be tempted to import the macros from the `log` crate.

For consistency, it would be good to have an option to enforce the use of the full path with clippy.

### Example

```rust
use log::error;

fn f() {
    error!("fatal error in f");
}
```

Could be written as:

```rust
fn f() {
    log::error!("fatal error in f");
}
```
